pkg/netpol/eval/eval_test.go

```diff
@@ -1870,3 +1870,105 @@ func TestAllParsedResourcesEvalTests(t *testing.T) {
 	runParsedResourcesEvalTests(t, examples.BANPWithNetPolV1FromParsedResourcesTest)
 	runParsedResourcesEvalTests(t, examples.ANPWithBANPFromParsedResourcesTest)
 }
+
+func TestDirPathEvalResults(t *testing.T) {
+	cases := []struct {
+		dir        string
+		sourcePod  string
+		sourceNs   string
+		destNs     string
+		destPod    string
+		protocol   string
+		port       string
+		evalResult bool
+	}{
+		{
+			dir:        "anp_demo",
+			sourceNs:   "gryffindor",
+			sourcePod:  "harry-potter",
+			destPod:    "luna-lovegood",
+			destNs:     "ravenclaw",
+			protocol:   "udp",
+			port:       "52",
+			evalResult: true,
+		},
+		{
+			dir:        "anp_test_6",
+			sourceNs:   "network-policy-conformance-slytherin",
+			sourcePod:  "draco-malfoy",
+			destPod:    "cedric-diggory",
+			destNs:     "network-policy-conformance-hufflepuff",
+			protocol:   "udp",
+			port:       "5353",
+			evalResult: false,
+		},
+		{
+			dir:        "anp_test_multiple_anps",
+			sourceNs:   "network-policy-conformance-ravenclaw",
+			sourcePod:  "luna-lovegood",
+			destPod:    "draco-malfoy",
+			destNs:     "network-policy-conformance-slytherin",
+			protocol:   "sctp",
+			port:       "9003",
+			evalResult: false,
+		},
+		{
+			dir:        "anp_with_np_and_banp_pass_test",
+			sourceNs:   "ns2",
+			sourcePod:  "pod1",
+			destPod:    "pod1",
+			destNs:     "ns1",
+			port:       "80",
+			evalResult: true,
+		},
+		{
+			dir:        "anp_with_np_pass_test",
+			sourceNs:   "ns2",
+			sourcePod:  "pod1",
+			destPod:    "pod1",
+			destNs:     "ns1",
+			port:       "8080",
+			evalResult: false,
+		},
+		{
+			dir:        "anp_banp_core_test",
+			sourceNs:   "network-policy-conformance-gryffindor",
+			sourcePod:  "harry-potter",
+			destPod:    "cedric-diggory",
+			destNs:     "network-policy-conformance-hufflepuff",
+			port:       "8080",
+			evalResult: true,
+		},
```

Comment:
yes, the names are of the relevant workloads in the manifests.
I mentioned that the input-dir does not contain pods so it will generate a pod matching the workload resource with the src + dst names.
e.g. the generated src pod takes the `metadata` of the workload with the `sourcePod` param name. (i.w. its name, namespace and labels)

Reply:
ok, so the test attribute should be `srcWorkload` or `dst`Workload`, and also better add some documentation explaining that for these tests `eval` will be called with pod names owned by these workloads (pods which are added by policy engine ).

Reply:
done